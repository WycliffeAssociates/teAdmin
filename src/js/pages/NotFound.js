import React, { Component } from 'react';
import {Button, Divider, Segment} from 'semantic-ui-react';
import {Link} from 'react-router-dom';

class NotFound extends Component {

  render() {
    return (
      <Segment raised textAlign="center">
                 This page does not exist.
        <Divider />
        <Link to="/">
          <Button color="green" content="Dashboard" />
        </Link>
      </Segment>
    );
  }
}

export default NotFound;
