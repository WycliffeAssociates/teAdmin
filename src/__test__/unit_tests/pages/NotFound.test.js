/* global it expect describe*/
import React from 'react';
import NotFound from '../../../js/pages/NotFound';
import renderer from 'react-test-renderer';
import {Router} from 'react-router';
import {createMemoryHistory} from 'history';

const history = createMemoryHistory('/notexist');

describe('NotFound Page', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Router history={history}>
        <div> <NotFound  />
        </div>
      </Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
