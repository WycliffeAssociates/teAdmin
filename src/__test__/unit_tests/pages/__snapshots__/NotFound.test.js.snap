// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page renders correctly 1`] = `
<div>
   
  <div
    className="ui raised center aligned segment"
  >
    This page does not exist.
    <div
      className="ui divider"
    />
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="ui green button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        Dashboard
      </button>
    </a>
  </div>
</div>
`;
